### PR TITLE
Dawnport vocations extinction and other dawnport fixes

### LINKED ISSUE #2092: FEATURE: Skill Conversion system for dawnport vocations
#### Describe your feature request
A skill conversion system for dawnport vocations.

#### Describe the solution you'd like
When a dawnport player change vocation his/er skills should change to fit the new vocation. It would block players to select eg druid and level up its ml and then change to knight and keeping the magic level from druid time.

#### Describe alternatives you've considered
None but a direct system to convert it

### LINKED ISSUE #2093: BUG: skill limit for dawnport characters
#### Describe the bug
The limit system from server, that should limit dawnport's players from level up skills, isnt working.
You're able to set a limit eg 20/20 (base skills/ml) and yet the player is able to level up it more than that.

#### How to Reproduce?
Configure the file properly `data/lib/others/dawnport_lib.lua` starting at line 197 for sorcerer vocation.
Create a dawnport character, if you did change the limit for only one vocation, select that vocation and keep training your skills, axe/sword/etc to see you going higher than the limit previously set.

#### Logs or errors
No errors

#### Expected behavior
The player shouldnt be able to rise his/er skills above the limit set

#### Environment
- OS (e.g. Windows): Debian 10
- OTServ Version (e.g. commit): develop
- Tibia Version (e.g. 12.52): 12.51

### LINKED ISSUE #2299: BUG: Mana bug when choosing 'rookgaard' at npc Inigo.
#### How to Reproduce?
Go to NPC Inigo and say "hi - rookgaard - yes - yes" and your mana will be bugged

#### Possible fix
at `data/lib/others/dawnport_lib.lua` at line 185 (On function rookgaardSetStats(player)) the math is not working properly, its returning negative value when the character have less then level 5. This math need to be changed.

#### Additional context
This issue have a possible fix on the PR #2297, if the PR is not merged i will create a new PR to fix this one, that's why this issue exists.

### PR DESCRIPTION
**Pull request by ElTeixe**
Removed dawnport vocations
-
- Its only purpose its handle the stats (hp/mana/cap) gain/loss for vocations with level <= 8.
- Its handled on source with 2 simple 'if' and avoids duplications, the headache trying to handle them switching between dawnport and main vocations and vice versa, scripted equipments/spells, between other stuffs.
- It was removed or adjusted everything related to dawnport vocations.

Spells
-
- Migrated to regular spells the special scripted spells used to handle dawnport vocations.
- (Bruise Bane, Magic Patch, Chill Out, Mud Attack, Arrow Call, Buzz and Scorch)

Equipments
-
- Migrated to regular unscripted equipments the scripted equipments used to handle dawnport vocations.
- (Scorcher, Chiller and Spellbook of the Novice.)

Vocations trial
-
- Fixed only removing x2 of each previous trial equipments when switching vocation.
- Fixed error when trying move into backpack an equipped item when there is no backpack.
- Added magic level and skill level conversion between vocations.
- Added quiver to paladin trial, adding simple arrows inside.

Vocations door
-
- Reworked and corrected the items removed on pass through the door.
- Added missing message on pass through the door.

Vocations reward
-
- Corrected some reward items.
- Added missing Adventurers Guild parchment.
- Added check of enough capacity and enough slots before able to take the reward.
- Added quiver to paladin rewards.

Dawnport tiles
-
- Added missing cure poison MoveEvent for dawnport outpost entrances tiles.
- Fixed vocations chest room exits and added missing message.

NPCs
-
Oressa:
- Added magic level and skill level conversion between vocations on choose final vocation.
- Reworked and corrected the items removed on choose vocation.

Inigo:
- Added condition of level <= 9 to ask about to go Rookgaard.
- Added magic level and skill level conversion to none vocation on go to Rookgaard.

Captain Dreadnought:
- Added missing transcripts.
- Added default town travel.
- Corrected teleport to town boat positions instead of town temple positions.
- Added force to drop or donate the excess of 500 gold coins before to sail a town.
- Customizable just editing towns table.

Other
-
Dawnport skill limit:
- Added skill limit to dawnport residents. Notifies about and limit it when reached.
- By default only limits magic level like Official (Sorcerer/Druid 20, Paladin 9 and Knight 4). Can be customized to another skills.

New Lua scripting functions:
- player:setMagicLevel(level[, manaSpent])
- player:setSkillLevel(skillType, level[, tries])

And other small edits like removed variables not in use, moved variables/functions from global to local, etc, etc that Im not gonna name them all.

NOTE: If magic level and skill levels conversion is not matching exactly Official, its due the multipliers at vocations.xml are not the same. 
- I didnt have enough time to figure them all, but for sure knight/elite knight manamultipliers should be 4.0 instead 3.0.
- Sorcerer/druid and paladins manamultipliers should be higher too but I dont have final exact results yet.

Closes #2092 Closes #2093 Closes #2299

### CODE REVIEW

**Comment by marcosvf132** [comment]
Fix #2299

**Review by dex-89 — CHANGES_REQUESTED** [review]
It looks like everything is fine. If you could improve the review I would be very grateful: unnecessary spaces and line length.

Besides - great job

**Comment by ElTeixe** [comment]
> It looks like everything is fine. If you could improve the review I would be very grateful: unnecessary spaces and line length.
> 
> Besides - great job

- About unnecessary spaces, you mean empty line jumps between 'ifs' and variables?
- About the lines length, I guessed that gonna be a complain, since I seen that practice at other scripts. I had that discussion many times at the job too, about how length must be.. 125 if I dont remember wrong its small IMHO, I dont like have half width screen empty hahah but I can adapt myself.

Confirm about, and I'll try correct it.

**Comment by dex-89** [comment]
> > It looks like everything is fine. If you could improve the review I would be very grateful: unnecessary spaces and line length.
> > Besides - great job
> 
> * About unnecessary spaces, you mean empty line jumps between 'ifs' and variables?
> * About the lines length, I guessed that gonna be a complain, since I seen that practice at other scripts. I had that discussion many times at the job too, about how length must be.. 125 if I dont remember wrong its small IMHO, I dont like have half width screen empty hahah but I can adapt myself.
> 
> Confirm about, and I'll try correct it.

All files to fix:
https://github.com/opentibiabr/otservbr-global/pull/2297/checks?check_run_id=1749599333

unnecessary spaces: `line contains only whitespace`
`data/events/scripts/player.lua:770:1: line contains only whitespace`

line lenght: `line is too long (140 > 120)`
`data/lib/others/dawnport_lib.lua:118:121: line is too long (122 > 120)`

File: src/luascript.cpp [hunk 1]
```diff
@@ -2587,6 +2587,9 @@ void LuaScriptInterface::registerFunctions()
 	registerMethod("Player", "getSkillTries", LuaScriptInterface::luaPlayerGetSkillTries);
 	registerMethod("Player", "addSkillTries", LuaScriptInterface::luaPlayerAddSkillTries);
 
+  registerMethod("Player", "setMagicLevel", LuaScriptInterface::luaPlayerSetMagicLevel);
```
Comment (slawkens) [inline src/luascript.cpp]: Use tabs instead of two spaces :P That apply to all your .cpp and .h changes

Comment (ElTeixe) [reply src/luascript.cpp]: I got noticed after commit the code, in my VS it looks right. I dont know why VS dont follow the same spacing/tabs defined on the file  instead use another and change it. I try correct that too.

Comment (dex-89) [reply src/luascript.cpp]: Try to use Notepad++ to edit and add in it:
image

**Review by dex-89 — APPROVED** [review]

**Review by majestyotbr — APPROVED** [review]

**Pull request merged**